pkg/plugins/platform/router.go

```diff
@@ -84,3 +86,45 @@ func StartDockerRouter() {
 		log.Fatalf("Could not start router: %v", err)
 	}
 }
+
+// GetCurrentRouterPorts retrieves the ports currently exposed on the router if it is running.
+func GetCurrentRouterPorts() []string {
+	var exposedPorts []string
+	label := map[string]string{"com.docker.compose.project": "ddevrouter"}
+	router, err := util.FindContainerByLabels(label)
+	if err == nil {
+		ports := router.Ports
+		for _, port := range ports {
+			if port.PublicPort != 0 {
+				exposedPorts = append(exposedPorts, fmt.Sprintf("%v:%v", port.PublicPort, port.PrivatePort))
+			}
+		}
+	}
+	return exposedPorts
+}
+
+// SetRouterPorts determines the router port configuration based on port definitions from
+// site containers and any current configuration from a running router.
+func SetRouterPorts(ports []string) []string {
```

Comment:
Currently this would maintain the existing ports. The intent is to keep port definitions from other sites in place without having to re-query every container for every running site. This likely falls apart if an existing external port gets remapped to a different internal port though.